src/Middleware/RequestThrottling/src/Internal/RequestQueue.cs

```diff
@@ -10,35 +10,37 @@ namespace Microsoft.AspNetCore.RequestThrottling.Internal
     internal class RequestQueue : IDisposable
     {
         private SemaphoreSlim _semaphore;
-        private object _waitingRequestsLock = new object();
-        public readonly int MaxConcurrentRequests;
-        public int WaitingRequests { get; private set; }
+        private int _waitingRequests;
 
-        public RequestQueue(int maxConcurrentRequests)
+        private readonly int _maxConcurrentRequests;
+        private readonly int _requestQueueLimit;
+
+        public RequestQueue(int maxConcurrentRequests, int requestQueueLimit)
         {
-            MaxConcurrentRequests = maxConcurrentRequests;
+            _maxConcurrentRequests = maxConcurrentRequests;
+            _requestQueueLimit = requestQueueLimit;
             _semaphore = new SemaphoreSlim(maxConcurrentRequests);
         }
 
-        public async Task EnterQueue()
-        {
-            var waitInQueueTask = _semaphore.WaitAsync();
+        public RequestQueue(int maxConcurrentRequests) : this(maxConcurrentRequests, 5000) { }
```

Comment:
Is this constructor only for test convenience? If so, replace it with a helper method in the test class. We don't want defaults hard-coded in places like this, or test helpers in production code.